### DISCUSSION TITLE
What API event can be used to modify the upkeep a Town pays?

### DISCUSSION THREAD

**Discussion by manul694** [TownyAdvanced/Towny#6014] [Q&A] [answered]
Is towny has an event that calling when town paying tax for territories?

**Comment by LlmDl** [comment]
There are transaction events that could be listened to, there is also an event thrown when the new-day happens. What are you looking to do exactly?

**Comment by manul694** [comment]
I want to set territories taxes depend of my own conditions

**Reply by LlmDl** [reply]
Tax or upkeep?

**Reply by manul694** [reply]
Upkeep.
So I found it because your comment, TownUpkeepCalculationEvent. Thank you very much

**Comment by LlmDl** [comment, marked as answer]
The TownUpkeepCalculationEvent is used to modify what Upkeep a town will pay.
